### PR TITLE
add async and typings with typescript

### PR DESCRIPTION
**Pull request by Rob9315**
This PR converts yggdrasil to typescript with all functions now being asynchronous, while still retaining the callback compatibility

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -1,95 +1,123 @@
...
-    delete options.token
-  } else {
-    options.token = options.token || uuid.v4()
+const Client = {
+  /**
+     * Attempts to authenticate a user.
+     * @param  {Object}   options Config object
+     * @param  {Function} cb      Callback
+     */
+  auth: async function (options, cb) {
```
Comment (rom1504) [inline lib/Client.js:11]: Avoid new Promise and instead convert the inner methods to promise

**Comment by rom1504** [comment]
Why the mix between typescript and javascript? Can't we use only javascript ?

**Comment by Rob9315** [comment]
will change things up, typescript because of automatic typings

**Review by rom1504 — CHANGES_REQUESTED** [review]
Can you use ts-standard ? Standard doesn't work with typescript

**Comment by rom1504** [comment]
What about the tests ? Not using typescript and async there ?

File: package.json [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
-}
+{
+  "name": "yggdrasil",
+  "version": "1.5.0",
+  "author": "Zeke Sonxx <zeke@zekesonxx.com>",
+  "description": "Mojang authentication (Yggdrasil) client",
+  "files": ["dist/**/*"],
+  "scripts": {
+    "test": "mocha -R spec",
+    "pretest": "npm run fix && npm run build",
```
Comment (rom1504) [inline package.json:9]: npm run fix shouldn't run on pretest and prepublish, only non run lint

**Comment by Kashalls** [comment]
Any updates?

File: package.json [hunk 3]
```diff
@@ -1,35 +1,44 @@
...
+  "version": "1.5.0",
+  "author": "Zeke Sonxx <zeke@zekesonxx.com>",
+  "description": "Mojang authentication (Yggdrasil) client",
+  "files": [
+    "dist/**/*"
+  ],
+  "scripts": {
+    "test": "mocha -R spec",
+    "pretest": "npm run lint && npm run build",
+    "prepublish": "npm run fix && npm run build",
```
Comment (rom1504) [inline package.json:12]: lint should be there, not fix

**Comment by rom1504** [comment]
I left a minor comment, otherwise looks good.
We'll probably remove the callback tests soon but for now let's keep them indeed.

**Review by rom1504 — APPROVED** [review]

**Pull request merged**